### PR TITLE
Week 6 Assignment

### PR DESCRIPTION
**Pull request by janewaitara**
On this Branch, I modified week 5 assignment to populate the Room Database and connected the UI to the data.
I used Coroutines for background processes.

### CODE REVIEW

File: Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/MovieApplication.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        lateinit var database: MovieDatabase
+
+        fun getAppContext(): Context = instance.applicationContext
+    }
+
+    override fun onCreate() {
+        instance = this
+        super.onCreate()
+
+        database = Room.databaseBuilder(this,
```
Comment (jenndotcodes) [inline Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/MovieApplication.kt:22]: Nice job not including the run queries on the main thread here.

File: Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/db/RoomDatabase.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import com.janewaitara.movieapp.model.Movie
+import com.janewaitara.movieapp.MovieApplication
+import com.janewaitara.movieapp.ui.movies.MovieViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+
+@Dao
+interface MovieDao{
```
Comment (jenndotcodes) [inline Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/db/RoomDatabase.kt:15]: Really excellent job on this class.

File: Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/db/RoomDatabase.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    private class MovieDatabaseCallback(private val scope: CoroutineScope)
+        : RoomDatabase.Callback(){
+        override fun onCreate(db: SupportSQLiteDatabase) {
+            super.onCreate(db)
+            scope.launch {
+                var movieDao = getDatabase()
+                    .movieDao()
+                //populating
+                movieDao.insertAllMovies(MovieViewModel.movieList)
```
Comment (jenndotcodes) [inline Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/db/RoomDatabase.kt:90]: Excellent work here inserting the movies.

File: Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/ui/login/LoginFragment.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    /**
+     * a fun to validate whether the login userName is empty or shorter than expected.
+     * Returns a Boolean
+     * */
+    private fun validateUserName(): Boolean{
+        val loginUserName = login_userNameWrapper.editText!!.text.toString().trim()
+
+        return when {
```
Comment (jenndotcodes) [inline Week 6/MovieAppW6/app/src/main/java/com/janewaitara/movieapp/ui/login/LoginFragment.kt:74]: Really nice job on the when statement here and doing the user validation.

**Review by jenndotcodes — COMMENTED** [review]
Jane, I certainly did enjoy reviewing this. This code is exceptionally clean and very professional. Excellent use of coroutines and coroutine scope, and the overall architecture of the app is very good. The commenting made the code easy to understand and very readable. This project looks professional and is exceptional. I am assigning 'Above and Beyond'. Keep up the great work!

**Comment by janewaitara** [comment]
> Jane, I certainly did enjoy reviewing this. This code is exceptionally clean and very professional. Excellent use of coroutines and coroutine scope, and the overall architecture of the app is very good. The commenting made the code easy to understand and very readable. This project looks professional and is exceptional. I am assigning 'Above and Beyond'. Keep up the great work!

Thank you Jenn.
I am trying my best to avoid spaghetti code

**Pull request merged**